lib/db/transactions.go

```diff
@@ -121,24 +162,33 @@ func (t readWriteTransaction) updateGlobal(gk, keyBuf, folder, device []byte, fi
 		// Inserted a new newest version
 		global = file
 	} else {
-		keyBuf = t.keyer.GenerateDeviceFileKey(keyBuf, folder, fl.Versions[0].Device, name)
-		if new, ok := t.getFileByKey(keyBuf); ok {
-			global = new
-		} else {
-			// This file must exist in the db, so this must be caused
-			// by the db being closed - bail out.
+		keyBuf, err = t.keyer.GenerateDeviceFileKey(keyBuf, folder, fl.Versions[0].Device, name)
+		if err != nil {
+			return nil, false, err
+		}
+		new, ok, err := t.getFileByKey(keyBuf)
+		if err != nil {
+			return nil, false, err
```

Comment:
Same here: I don't see a reason to not return `keyBuf` and thus can combine with if below.

And I'd return `keyBuf` instead of nil on error also in all the other instances. I'll stop commenting to have less conversations to hide if fixed/dismissed.